Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+            _historicalPortfolioCapacity = new RollingWindow<decimal>(30);
+            _previousInputTime = DateTime.MinValue;
+
+            orderEventProvider.NewOrderEvent += (sender, orderEvent) =>
+            {
+                var symbol = orderEvent.Symbol;
+
+                var configs = _subscriptionDataConfigProvider.GetSubscriptionDataConfigs(symbol);
+                if (configs.IsNullOrEmpty())
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: Could not find {symbol} in Portfolio");
+                }
+
+                try
+                {
+                    var consolidator = default(IDataConsolidator);
+                    var security = _portfolio.Securities[symbol];
+                    BaseData lastData = security.Cache.GetData<TradeBar>();
+
+                    // For high resolution data, create a consolidator that will compute
+                    // the AUM Capacity after a 5-minute period bar is closed
+                    foreach (var config in configs.Where(x => x.Resolution < Resolution.Hour))
+                    {
+                        if (config.Type.IsAssignableFrom(typeof(TradeBar)))
+                        {
+                            consolidator = new TradeBarConsolidator(TimeSpan.FromMinutes(5));
+                        }
+
+                        if (config.Type.IsAssignableFrom(typeof(Tick)) &&
+                            config.TickType == TickType.Trade)
+                        {
+                            consolidator = new TickConsolidator(TimeSpan.FromMinutes(5));
+                            lastData = security.Cache.GetData<Tick>();
+                        }
+
+                        if (consolidator == null)
+                        {
+                            continue;
+                        }
+
+                        // Warm up the consolidator to mark the begging of the period
+                        if (lastData != null)
+                        {
+                            consolidator.Update(lastData);
+                        }
+
+                        config.Consolidators.Add(consolidator);
+                        consolidator.DataConsolidated += (s, data) =>  Update(s, data, orderEvent.OrderId);
+                        return;
+                    }
+
+                    // For low resolution, use the last data available
+                    if (lastData != null)
+                    {
+                        Update(sender, lastData, orderEvent.OrderId);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: {e}");
+                }
+            };
+        }
+
+        /// <summary>
+        /// Updates the AUM Capacity with the latest data available after a order fill
+        /// </summary>
+        /// <param name="sender">For high resolution, sender object represents the data consolidator</param>
```

Comment:
Is this comment correct? This [line](https://github.com/QuantConnect/Lean/pull/3573/files#diff-6b9a41074e2cd8dc4fdbf36e8b798a2fR124) appears to be passing the `sender` object provided via `IOrderEventProvider.NewOrderEvent` which is likely the transaction handler, correct?